File: MyMedTimer/Utilities/TimeFormatting.swift
import Foundation

enum TimeFormatting {

    static func countdown(_ interval: TimeInterval) -> String {
        if interval < 0 { return "overdue" }
        if interval == 0 { return "now" }
        if interval < 60 { return "<1m" }

        let totalMinutes = Int(interval) / 60
        let hours = totalMinutes / 60
        let minutes = totalMinutes % 60

        if hours > 0 {
            return "\(hours)h \(minutes)m"
        }
        return "\(minutes)m"
    }

    static func timeOfDay(hour: Int, minute: Int) -> String {
        var components = DateComponents()
        components.hour = hour
        components.minute = minute

        let formatter = DateFormatter()
        formatter.dateFormat = "h:mm a"

        let calendar = Calendar.current
        let date = calendar.date(from: components) ?? Date()
        return formatter.string(from: date)
    }
}

File: android/core/common/src/main/kotlin/com/nateb/mymedtimer/common/TimeFormatting.kt
package com.nateb.mymedtimer.common

import java.time.Instant
import java.time.LocalTime
import java.time.ZoneId
import java.time.format.DateTimeFormatter

object TimeFormatting {

    fun countdown(intervalSeconds: Long): String {
        if (intervalSeconds < 0) return "overdue"
        if (intervalSeconds == 0L) return "now"
        if (intervalSeconds < 60) return "<1m"

        val totalMinutes = intervalSeconds / 60
        val hours = totalMinutes / 60
        val minutes = totalMinutes % 60

        return if (hours > 0) {
            "${hours}h ${minutes}m"
        } else {
            "${minutes}m"
        }
    }

    fun timeOfDay(hour: Int, minute: Int): String {
        val time = LocalTime.of(hour, minute)
        val formatter = DateTimeFormatter.ofPattern("h:mm a")
        return time.format(formatter)
    }

    fun shortTime(instant: Instant): String {
        val time = instant.atZone(ZoneId.systemDefault()).toLocalTime()
        val formatter = DateTimeFormatter.ofPattern("h:mm a")
        return time.format(formatter)
    }
}
